Assets/Scripts/Combat/Abilities/Bonk.cs

```diff
@@ -0,0 +1,170 @@
+using Assets.Scripts.Combat;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+namespace Combat.Abilities
+{
+    public class Bonk : Ability
+    {
+
+        private enum Phase { Inactive, Approaching, Bonking, Disengaging }
+        private Phase CurrentPhase = Phase.Inactive;
+
+        private Timer Timer;
+        private const float ApproachingDuration = 2.5f;
+        private const float BonkingDuration = 3.0f;
+        private const float DisengagingDuration = 1.5f;
+
+        private const float Damage = 25;
+
+        private Vector3 InitialPosition;
+        private GameObject Victim;
+        private Vector3 AttackingPosition;
+        private readonly float SpaceBetweenBonk = 2.5f;
+
+        // Maybe there's a better way of finding this out dynamically...?
+        [SerializeField]
+        private bool BelongsToAlly;
+
+
+        public new void Start()
+        {
+            base.Start();
+            Timer = GetComponent<Timer>();
+
+            TargetSchema = new TargetSchema(
+                1,
+                BelongsToAlly ? CombatantType.Enemy : CombatantType.Ally,
+                SelectorType.Number);
+        }
+
+        private void Update()
+        {
+            if (CurrentPhase == Phase.Inactive)
+                return;
+
+            if (CurrentPhase == Phase.Approaching)
+                ApproachingUpdate();
+
+            if (CurrentPhase == Phase.Bonking)
+                BonkingUpdate();
+
+            if (CurrentPhase == Phase.Disengaging)
+                DisengagingUpdate();
+        }
+
+        private void ApproachingUpdate()
+        {
+            if (Timer.IsInProgress())
+            {
+                float t = Timer.GetProgress() / ApproachingDuration;
+                transform.parent.gameObject.transform.position = Vector3.Lerp(
+                    InitialPosition,
+                    AttackingPosition,
+                    t);
+                // while this is happening, play George's walking animation
+            }
+
+            else if (Timer.IsFinished())
+            {
+                StartBonkPhase();
+            }
+        }
+
+        private void DisengagingUpdate()
+        {
+            if (Timer.IsInProgress())
+            {
+                float t = Timer.GetProgress() / DisengagingDuration;
+                transform.parent.gameObject.transform.position = Vector3.Lerp(
+                    AttackingPosition,
+                    InitialPosition,
+                    t);
+            }
+
+            else if (Timer.IsFinished())
+            {
+                EndAbility();
+            }
+        }
+
+        private void BonkingUpdate()
+        {
+            if (Timer.IsInProgress())
+            {
+                // Empty for now
```

Comment:
What is this going to be used for?

Reply:
I left the skeleton as is for this Update helper. If we need to add any more logic to the ability, this is a placeholder that could help someone else coming into the code know where to put it